src/service.js

```diff
@@ -177,7 +190,18 @@ class Service {
   }
 
   patch (id, data, params) {
-    params.query = params.query || {};
+    const query = params.query || {};
+    const patchQuery = {};
+
+    // Account for potentially modified data
```

Comment:
@daffl probably. That would be easier to trace and also the way I would normally do it in my own code. However, it **might** be more expensive an operation than just doing what you are doing in memory. Especially if patching a lot of objects. We'd almost need to test with 10k documents or something.

A common use case for patches like that are database migrations so large numbers would be common.